bpf/tests/nat_test.h

```diff
@@ -99,3 +101,52 @@ void test_snat_v4_track_local() {
     assert(!snat_v4_track_local(&ctx, &tuple, &state, NAT_DIR_EGRESS, 0,
                                 &target));
 }
+
+// Create a fake map.
+HASHMAP(struct ipv4_ct_tuple, struct ipv4_nat_entry) ipv4_ct_tuple_map;
+
+// Define the compare_func
```

Comment:
They cannot be mvoed into a lib because the cmp and hash functions actually require a length which cannot be passed by  arguments. Since we cannot know the length of the key (the type of the key is not confirmed until the map is intialized) in advance, so the cmp and hash functions have to be wrapped once the type of the key is confirmed.